src/Uno.UI/UI/Xaml/Controls/Repeater/ViewManager.cs

```diff
@@ -0,0 +1,984 @@
+					}
+
+					int countChange = newCount - oldCount;
+					if (countChange != 0)
+					{
+						// countChange > 0 : countChange items were added
+						// countChange < 0 : -countChange  items were removed
+						var children = m_owner.Children;
+						for (int i = 0; i < children.Count; ++i)
+						{
+							var element = children[i];
+							var virtInfo = ItemsRepeater.GetVirtualizationInfo(element);
+							var dataIndex = virtInfo.Index;
+
+							if (virtInfo.IsRealized)
+							{
+								if (dataIndex >= oldStartIndex + oldCount)
+								{
+									UpdateElementIndex(element, virtInfo, dataIndex + countChange);
+								}
+							}
+						}
+
+						EnsureFirstLastRealizedIndices();
+						m_lastRealizedElementIndexHeldByLayout += countChange;
+					}
+
+					break;
+				}
+
+				case NotifyCollectionChangedAction.Remove:
+				{
+					var oldStartIndex = args.OldStartingIndex;
+					var oldCount = (int)(args.OldItems.Count);
+					var children = m_owner.Children;
+					for (int i = 0; i < children.Count; ++i)
+					{
+						var element = children[i];
+						var virtInfo = ItemsRepeater.GetVirtualizationInfo(element);
+						var dataIndex = virtInfo.Index;
+
+						if (virtInfo.IsRealized)
+						{
+							if (virtInfo.AutoRecycleCandidate && oldStartIndex <= dataIndex && dataIndex < oldStartIndex + oldCount)
+							{
+								// If we are doing the mapping, remove the element who's data was removed.
+								m_owner.ClearElementImpl(element);
+							}
+							else if (dataIndex >= (oldStartIndex + oldCount))
+							{
+								UpdateElementIndex(element, virtInfo, dataIndex - oldCount);
+							}
+						}
+					}
+
+					InvalidateRealizedIndicesHeldByLayout();
+					break;
+				}
+
+				case NotifyCollectionChangedAction.Reset:
+					// If we get multiple resets back to back before
+					// running layout, we dont have to clear all the elements again.         
+					if (!m_isDataSourceStableResetPending)
+					{
+						// There should be no elements in the reset pool at this time.
+						MUX_ASSERT(m_resetPool.IsEmpty);
+
+						if (m_owner.ItemsSourceView.HasKeyIndexMapping)
+						{
+							m_isDataSourceStableResetPending = true;
+						}
+
+						// Walk through all the elements and make sure they are cleared, they will go into
+						// the stable id reset pool.
+						var children = m_owner.Children;
+						for (int i = 0; i < children.Count; ++i)
+						{
+							var element = children[i];
+							var virtInfo = ItemsRepeater.GetVirtualizationInfo(element);
+							if (virtInfo.IsRealized && virtInfo.AutoRecycleCandidate)
+							{
+								m_owner.ClearElementImpl(element);
+							}
+						}
+					}
+
+					InvalidateRealizedIndicesHeldByLayout();
+
+					break;
+			}
+		}
+
+		void EnsureFirstLastRealizedIndices()
+		{
+			if (m_firstRealizedElementIndexHeldByLayout == FirstRealizedElementIndexDefault)
+			{
+				// This will ensure that the indexes are updated.
+				var element = GetElementIfAlreadyHeldByLayout(0);
```

Comment:
Codacy found an issue: [Remove the unused local variable 'element'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5863680)